Modelica/Constants.mo

```diff
@@ -16,11 +16,11 @@ package Constants
 
   // Machine dependent constants
   final constant Real eps=ModelicaServices.Machine.eps
-    "Biggest number such that 1.0 + eps = 1.0";
+    "The difference between 1 and the least value greater than 1 that is representable in the given floating point type";
```

Comment:
Could conversion annotations be used to avoid breaking things when changing this constant by a factor of 2?

Reply:
The default ModelicaServices that some tools used had eps=1e-15, so we are changing it by a factor of either 0.2 or 2 - which seems to be impossible to handle automatically.

In many cases it was just used to guard against too small (or basically it is a test of >0 and I don't see that as really strict), and in others it doesn't seem that strict due to additional factors, and in some cases it was wrong.

In MSL it is used in:
- Modelica.Electrical.QuasiStatic.SinglePhase.Basic.Admittance - small test
- Modelica.Media.Air.MoistAir.BaseProperties using 100*eps
- Modelica.ComplexBlocks.ComplexMath.Bode - small test
- Modelica.Blocks.Sources.CombiTimeTable - small test
- Modelica.ComplexBlocks.Sources.ComplexRampPhasor - broken, but new definition was intended
and 173 more classes

The interesting one is ComplexRampPhasor, which uses 
```
    if time < startTime then magnitude1 else 
    if time < (startTime + max(duration,eps)) then
```
That is broken if duration is too short and startTime too large (so that startTime+eps is the same as startTime) and the intent was likely something like:
```
    if time < startTime then magnitude1 else 
    if time < (startTime + max(duration, eps*abs(startTime))) then
```
(This is one of the case where DBL_EPS is intended; you want to ensure that you get a larger number.) However, being that strict is numerically messy for solvers so something like:
```
    if time < startTime then magnitude1 else 
    if time < (startTime + max(duration, 100*eps*abs(startTime))) then
```
would make more sense, and then it should be a bound parameter instead of repeating it... (or perhaps you should limit duration instead?)
Luckily in most cases this does not matter, since duration isn't that short.

Reply:
> The default ModelicaServices that some tools used had eps=1e-15, so we are changing it by a factor of either 0.2 or 2 - which seems to be impossible to handle automatically.

I'm not talking about the default.  I'm talking about the definition, and how changing the definition would impact models relying on a correct value for this constant.

Reply:
Checking directly on eps should nearly always be wrong. One should always check with a factor, such as 100*eps. Therefore, from my point of view the exact definition of eps doesn't matter much (especially with a factor of 2 or 0.2 for different definitions).  It seems practical to just use the constant from the C-standard.

For inf and small this is different. I think the definition that inf and -inf and small and -small must be representable on a machines simplifies its usage, because it is less likely that an unexperienced user makes a mistake. 
If one changes this definition now, one would have to check whether current code is still correct (e.g. Modelica.Math.Vectors.norm checks on p >=Modelica.Constants.inf). If there would be checks for -inf or -small, some code might wrong with the proposed change

Reply:
Martin: note that my proposed changes were more "clarifications" of this intent than any change.

Specifically inf is finite despite the name (so it copies definition of DBL_MAX - which is maximum).

The small is also finite, and importantly a normalized number which makes things easier - whereas previous definition of "smallest" seemed to allow non-normalized numbers that are weird (and might not work with some floating-point settings).

For eps the main change is that the previous text looked like it was defining eps/2 (or eps/2*(1+eps) ) I think; not eps - but that was clearly not intended.

I agree that you normally use a factor with that, e.g. 100*eps, but it is also a matter of knowing that it is relative, and that relative part was also missing in ComplexRampPhasor.

Reply:
I think all of these changes are good even though they are backwards incompatible.  Hence, if applied we need to make sure this is communicated so that tools can adapt to the new definitions.  System Modeler, for example, will have to modify its `eps`. We won't need to make any change for `small`, however, since we have a slightly wrong implementation today by using a normalized number.

Reply:
Unless there are any objections I will resolve this.